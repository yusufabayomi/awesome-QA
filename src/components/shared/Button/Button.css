.button {
  display: inline-block;
  padding: 0.8rem 0;
  border: 0;
  font-weight: 600;
  border-radius: 3px;
  font-size: 14px;
  color: #ffffff;
}

.btn-green {
  background-color: #52ab82;
  border-bottom: 3px solid #198754;
}

.btn-red {
  background-color: #f96c79;
  border-bottom: 3px solid #dc3545;
}

.btn-blue {
  background-color: #4895da;
  border-bottom: 3px solid #1c7ed6;
}
