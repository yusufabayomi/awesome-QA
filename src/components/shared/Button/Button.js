import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import './Button.css';

const propTypes = {
  buttonText: PropTypes.string.isRequired,
  buttonColor: PropTypes.oneOf(['blue', 'red', 'green']).isRequired,
  emitEvent: PropTypes.func,
};

const defaultProps = {
  icon: null,
};

const Button = ({ buttonText, buttonColor, icon, emitEvent }) => {
  return (
    <button data-testid='button' className={`button btn-${buttonColor} w-100`} onClick={emitEvent}>
      {icon && <FontAwesomeIcon icon={icon} data-testid='buttonIcon' />} {buttonText}
    </button>
  );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
