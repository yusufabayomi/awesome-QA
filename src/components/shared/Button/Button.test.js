import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Button from './Button';
import { faSort } from '@fortawesome/free-solid-svg-icons';

afterEach(cleanup);

describe('Renders Button Component', () => {
  const props = {
    buttonText: 'Sort Questions',
    buttonColor: 'blue',
    emitEvent: jest.fn(),
  };

  it('renders correctly', () => {
    const { asFragment } = render(<Button {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a FontAwesomeIcon when icon prop is defined', () => {
    const { getByTestId } = render(<Button {...props} icon={faSort} />);
    expect(getByTestId('buttonIcon')).toBeTruthy();
  });

  it('should emit a callback on click event', () => {
    const { getByTestId } = render(<Button {...props} />);
    fireEvent.click(getByTestId('button'));
    expect(props.emitEvent).toHaveBeenCalled();
  });
});
