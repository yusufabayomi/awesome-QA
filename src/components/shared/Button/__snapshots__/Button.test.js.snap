// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button Component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button btn-blue w-100"
    data-testid="button"
  >
     Sort Questions
  </button>
</DocumentFragment>
`;
